src/controllers/trialController.ts

```diff
@@ -59,10 +65,18 @@ export default class TrialController {
       withValidatedOrValidationError(
         TrialId.decode(req.params.trialId),
         (trialId) =>
-          withValidatedOrValidationError(
-            NonEmptyString.decode(user.fiscal_code),
-            (userId) => this.trialService.getSubscription(userId, trialId)
-          )
+          trialId !== IO_WALLET_TRIAL_ID || FF_IO_WALLET_TRIAL_ENABLED
```

Comment:
why this control. If it is the same it should check it right? In general I would just leave the control on the feature flag

Reply:
we can't remove the check on the `trialId` because the endpoint is for all trialIds not just for the wallet

Reply:
Ahh ok so we are saying that in the case of Wallet the check is not done. How about adding a comment?

Reply:
sure